### PR TITLE
Refactoring with Redux + Feature filter sort

### PR DESCRIPTION
**Pull request by baktun14**
-redux structure
-packages filtering + sorting with reselect
-bug correction

### CODE REVIEW

File: src/reducers/rootReducer.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { combineReducers } from "redux";
+import appReducer from "../reducers/addReducer";
```
Comment (Redm4x) [inline src/reducers/rootReducer.ts]: Is `addReducer` supposed to be `appReducer` ?

File: package.json [hunk 2]
```diff
@@ -22,6 +22,10 @@
     "react-dom": "15.4.2",
     "react-ga": "2.1.2",
     "react-intl": "2.2.3",
+    "react-redux": "^5.0.4",
```
Comment (Redm4x) [inline package.json]: Could you use fixed versions like the other packages ?

**Review by Redm4x — CHANGES_REQUESTED** [review]

File: src/reducers/rootReducer.ts (hunk 1 above)
Comment (baktun14) [reply src/reducers/rootReducer.ts]: true true

**Comment by baktun14** [comment]
Should be good @Redm4x

**Review by Redm4x — APPROVED** [review]

**Pull request merged**